src/me/drton/jmavsim/SimpleSensors.java

```diff
@@ -221,6 +261,47 @@ public void update(long t) {
             gpsCurrent.time = System.currentTimeMillis() * 1000;
             gps = gpsDelayLine.getOutput(t, gpsCurrent);
         }
+
+        if (t > flowLast + flowInterval) {
+            flowUpdated = true;
+            flowData = new FlowData();
+
+            flowData.integration_time = (t - flowLast) * 1000;
+            flowLast = t;
+
+            flowData.distance = getSonarDist();
+            double ground_distance = getGroundDistance();
+
+            // Translational flow
+            Matrix3d rot = new Matrix3d(object.getRotation());
+            Vector3d flow3d = new Vector3d(object.getVelocity());
+            flow3d.scale(- 1 / ground_distance);
+            rot.transpose();
+            rot.transform(flow3d);
+
+            // Rotational flow
+            flow3d.x += object.getRotationRate().x;
```

Comment:
What are the units here? Does this all just work out?

Reply:
It is all rad/s.

Looking back at it, I am more surprised by the fact that x and y are not swapped (either here or in the block above for translational flow)